libminifi/include/core/logging/internal/LogCompressorSink.h

```diff
@@ -57,26 +58,44 @@ class LogCompressorSink : public spdlog::sinks::base_sink<std::mutex> {
   ~LogCompressorSink() override;
 
   template<class Rep, class Period>
-  std::unique_ptr<io::InputStream> getContent(const std::chrono::duration<Rep, Period>& time, bool flush = false) {
+  std::vector<std::unique_ptr<io::InputStream>> getContent(const std::chrono::duration<Rep, Period>& time, bool flush = false) {
     if (flush) {
       cached_logs_.commit();
       compress(true);
     }
-    LogBuffer compressed;
-    if (!compressed_logs_.tryDequeue(compressed, time) && flush) {
-      return createEmptyArchive();
+
+    std::vector<std::unique_ptr<io::InputStream>> log_segments;
+    const auto segment_count = compressed_logs_.itemCount();
+    for (size_t i = 0; i < segment_count; ++i) {
+      LogBuffer compressed;
+      if (!compressed_logs_.tryDequeue(compressed, time) && flush) {
+        break;
```

Comment:
I don't understand the purpose of checking `flush` here (I didn't understand it before this change either, but it got more complex).  As far as I can see, we only call this with `flush = true`.

Can we remove the `flush` argument?  If not, then I think the logic needs to be made clearer, either by a comment or some other way.

Reply:
You are right, there is no reason to keep that argument at the moment, removed in 84abc4df7100b28695cb75a753bb9df3f64a0620